Sources/ParseSwift/Authentication/3rd Party/ParseFacebook.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  ParseFacebook.swift
+//  ParseSwift
+//
+//  Created by Abdulaziz Alhomaidhi on 3/18/21.
+//  Copyright © 2021 Parse Community. All rights reserved.
+
+import Foundation
+#if canImport(Combine)
+import Combine
+#endif
+
+// swiftlint:disable line_length
+
+/**
+ Provides utility functions for working with Facebook User Authentication and `ParseUser`'s.
+ Be sure your Parse Server is configured for [sign in with Facebook](https://docs.parseplatform.org/parse-server/guide/#configuring-parse-server-for-sign-in-with-facebook).
+ For information on acquiring Facebook sign-in credentials to use with `ParseFacebook`, refer to [Facebook's Documentation](https://developers.facebook.com/docs/facebook-login/limited-login.
+ */
+public struct ParseFacebook<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Facebook authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case token
+
+        /// Properly makes an authData dictionary with the required keys.
+        /// - parameter facebookId: Required id for the user.
+        /// - parameter authToken: Required identity token for the user.
+        /// - returns: Required authData dictionary.
+        func makeDictionary(facebookId: String,
```

Comment:
Can you change these keys to match facebooks API? `facebookId` is okay, but `authToken` should be `accessToken`. I'm assuming you left this over from the `ParseApple` copy. Luckily, the newer key for limited login called `authenticationToken` described https://github.com/parse-community/parse-server/pull/7219 which uses the same key for the server, `token`. This means you should change all `authToken` to `authenticationToken`. You will need to propagate this change through the rest of ParseFacebook, comments, and the test cases.

This means, you also need to add a new field to your code for `accessToken` is this is used for Facebooks graph API. It looks the developer should either provide `accessToken` or `token`, but not both at the same time.

Reply:
keys have been changed so they match Facebook's API, they are: `id`, `expirationDate`, `accessToken` and `authenticationToken`